src/nxdoc/nuxeo-server/administration/maintenance/garbage-collecting-orphaned-blobs.md

```diff
@@ -0,0 +1,174 @@
+   }
+}
+```
+Observe `repository.default.queryBlobKeys` equals **true**.
+
+If a failure occurs during the migration, it will stop and the capability will not be acquired. Causes will have to be identified and fixed by analyzing the server logs. If you run a server with a version greater than LTS 2021-HF02, the `ecm:blobKeys` is computed whenever a document is created/updated even while the migration is running.
+
+#### Supported Blob Provider implementations
+
+This GC implementation only works with Blob Providers extending [BlobStoreBlobProvider](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/blob/BlobStoreBlobProvider.html) which are:
+- [S3BlobProvider](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/blob/s3/S3BlobProvider.html) (when using [amazon-s3-online-storage](https://connect.nuxeo.com/nuxeo/site/marketplace/package/amazon-s3-online-storage))
+- [LocalBlobProvider](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/blob/LocalBlobProvider.html)
+
+{{#> callout type='note' heading='2021 Default Blob Provider'}}
+In LTS 2021, the default Blob Provider is [DefaultBinaryManager](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/blob/binary/DefaultBinaryManager.html). This GC implementation is therefore not supported out of the box in LTS 2021. You can switch to [LocalBlobProvider](https://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/core/blob/LocalBlobProvider.html) to benefit from this GC implementation with this configuration property:
+```
+nuxeo.core.binarymanager=org.nuxeo.ecm.core.blob.LocalBlobProvider
+```
+See [NXP-31876](https://jira.nuxeo.com/browse/NXP-31876).
+{{/callout}}
+
+### Disablement
+
+Immediate Garbage Collection is enabled by default. You can disable it with the following configuration property:
+
+```
+nuxeo.bulk.action.blobGC.enabled=false
+```
+
+## Full Garbage Collection
+
+Since [LTS 2021-HF38]({{page page='nuxeo-server-release-notes-2021-38'}}) (see [NXP-28565](https://jira.nuxeo.com/browse/NXP-28565)), a new Full GC implementation is available to scan your blob store in order to detect and delete the blobs that are no longer referenced in your repository.
+
+This Full GC leverages the [Bulk Action Framework]({{page page='bulk-action-framework'}}). Like other bulk actions, the following configuration properties can be tweaked to fit your environment:
+```
+nuxeo.bulk.action.garbageCollectOrphanBlobs.defaultConcurrency=2
+nuxeo.bulk.action.garbageCollectOrphanBlobs.defaultPartitions=4
+```
+
+Please see the dedicated [Blobs Management Rest endpoint]({{page space='rest-api' page='blobs-endpoint#garbage-collect-documentand39s-blobs'}}) to invoke and monitor a Blob Full GC.
+
+{{#> callout type='info' }}
+Despite this implementation is designed to scale on big volumes of data, it will necessarily take some time to fully garbage collect a repository referencing a certain amount of blobs.
```

Comment:
```suggestion
Despite this implementation being designed to scale on big volumes of data, it will necessarily take some time to fully garbage collect a repository referencing a certain amount of blobs.
```

Reply:
Do we have an idea about the order of magnitude? I suppose we have some numbers from our benchmarks, and I'm sure customers will start asking.

Reply:
yes we do see https://benchmarks.nuxeo.com/continuous/platform-311/index.html#fullgc
198443ms for GC 12876 blobs